presto-mongodb/src/main/java/io/prestosql/plugin/mongodb/MongoMetadata.java

```diff
@@ -153,34 +157,36 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
     @Override
     public void createTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, boolean ignoreExisting)
     {
-        mongoSession.createTable(tableMetadata.getTable(), buildColumnHandles(tableMetadata));
+        MongoTableHandle table = new MongoTableHandle(tableMetadata.getTable());
+        mongoSession.createTable(table, buildColumnHandles(tableMetadata));
     }
 
     @Override
     public void dropTable(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
         MongoTableHandle table = (MongoTableHandle) tableHandle;
-
-        mongoSession.dropTable(table.getSchemaTableName());
+        mongoSession.dropTable(table);
     }
 
     @Override
     public void renameTable(ConnectorSession session, ConnectorTableHandle tableHandle, SchemaTableName newTableName)
     {
-        throw new UnsupportedOperationException();
+        MongoTableHandle table = (MongoTableHandle) tableHandle;
+        mongoSession.renameTable(table, newTableName);
```

Comment:
Revert this change because it's unrelated to the case insensitive commit. I would recommend sending another PR with the test.